impl/src/main/java/org/ehcache/internal/store/disk/DiskStore.java

```diff
@@ -1041,9 +1037,18 @@ public void remove() {
       Serializer<Element> elementSerializer = serializationProvider.createSerializer(Element.class, storeConfig.getClassLoader());
       Serializer<Object> objectSerializer = serializationProvider.createSerializer(Object.class, storeConfig.getClassLoader());
       
+      // todo: This should be enforced at the type system
+      Object identifier;
+      if(storeConfig instanceof PersistentStoreConfiguration<?, ?, ?>) {
+        identifier = ((PersistentStoreConfiguration) storeConfig).getIdentifier();
+      } else {
+        throw new IllegalArgumentException("Store.Configuration for DiskStore should implement Store.PersistentStoreConfiguration");
```

Comment:
Don't we want to allow the disk store to work with a non-persistent store configuration if it's configured in temp_swap mode?